style/servo/selector_parser.rs

```diff
@@ -238,7 +250,11 @@ impl PseudoElement {
             },
             PseudoElement::Backdrop |
             PseudoElement::DetailsSummary |
-            PseudoElement::Marker  => PseudoElementCascadeType::Lazy,
+            PseudoElement::Marker |
+            PseudoElement::ServoTextControlInnerEditor |
+            PseudoElement::ServoTextControlInnerContainer |
+            PseudoElement::ServoTextControlPlaceholder |
+            PseudoElement::ServoInputColorSwatch => PseudoElementCascadeType::Lazy,
```

Comment:
As these are standardised pseudo elements from a CSS spec, I don't think they should be `-servo` prefixed.

Reply:
These private pseudo selectors for UA stylesheet, I have defined `::placeholder` and `::color-swatch` as well.

Reply:
Why does the UA stylesheet need a separate selector? Can't it just use the standard one?

Reply:
> Why does the UA stylesheet need a separate selector? Can't it just use the standard one?

AFAICT this is the standard approach in Blink, WebKit and Gecko.

Typically a UA shadow DOM contains more elements than the number of web-exposed pseudo elements. Therefore we are introducing private selectors for them.

As background, we tried to not involve private selectors but adding a stylesheet directly into the UA shadow DOM. It causes a major performance regression, though, and got reverted.